dialer.go

```diff
@@ -111,6 +111,12 @@ func (d *dialer) DialStream(dial DialFN) (Stream, error) {
 
 	// TODO: support pooling of connections (i.e. keep multiple physical connections in flight)
 	if current == nil || idsExhausted || idled {
+		if current != nil {
+			// Need to run this on a goroutine because we call back to sessionClosed()
```

Comment:
`markDefunct` no longer calls `sessionClosed`. It still aquires the session's lock, but the critcal sections the lock protects requires to be very short anyway.

Reply:
It does, but indirectly. Dialer creates sessions with an beforeClose callback that calls sessionClosed. I'll update the comment.